pkg/member/member_control.go

```diff
@@ -136,6 +144,34 @@ func (m *memberControl) AddMemberAsLearner(ctx context.Context) error {
 	return nil
 }
 
+// AddLearnerWithRetry add a new member as a learner with exponential backoff.
+func (m *memberControl) AddLearnerWithRetry(ctx context.Context, retrySteps int, dataDir string) error {
+	backoff := miscellaneous.CreateBackoff(RetryPeriod, retrySteps)
+
+	if err := retry.OnError(backoff, func(err error) bool {
+		return err != nil
+	}, func() error {
+		// Additional safety check before adding a learner
+		if _, err := os.Stat(dataDir); err == nil {
```

Comment:
This code has an issue.
You can try playing around with using errors in if and else [here](https://go.dev/play/p/yNP7RmoCw4G).

Reply:
Hey @unmarshall ,
this is want I want to do: 
```go
func main() {
	if _, err := os.Stat("/tmp/tringo/bingo"); err == nil {
		fmt.Println("file is found")
	} else if !os.IsNotExist(err) {
		fmt.Printf("error other than not exist: %v", err)
	} else {
		fmt.Printf("file not found %v", err)
	}
}
```
to return `err` only when `err !=data dir not exist` 
if `err==data dir not exist` then I want to continue to add a learner

Reply:
ok, then just move out this logic into a separate function and write tests for it

Reply:
added the unit tests. PTAL